### PR TITLE
Add !immortal command

### PR DESCRIPTION
**Pull request by ibm2431**
Who wants to live forever?

Well, with this command we can be princes of the universe!

So while we're fighting to survive, we never fall to the darkest powers!

**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Review by zach2good — APPROVED** [review]
Cool!

**Comment by TeoTwawki** [comment]
Q: does unkillable prevent doom death and other instakills? :thinking: 

unkillable style instead of massive regen and 2hjhrs is legit how retail has their GMs not die. Thank you for thinking of this, it never occurred to me that flag would work on a player entity, I've been using a stripped down godmode instead for years when I want to test things.

**Review by TeoTwawki — APPROVED** [review]

**Comment by zach2good** [comment]
image

File: scripts/commands/immortal.lua [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        targ = player
+    end
+
+    if not targ:isNPC() then
+        if targ:isAlive() then
+      
+            local immortal = false
+            if targ:isPC() then
+                if targ:getCharVar("Immortal") == 1 then
+                    targ:delStatusEffect(tpz.effect.ERGON_MIGHT)
```
Comment (TeoTwawki) [inline scripts/commands/immortal.lua]: um didn't catch the use of a status effect my first look over. lets not do that. 

Lets just used addStatusEX to make its a duplicate of brew effect instead, for the lols :+1:

**Review by TeoTwawki — CHANGES_REQUESTED** [review]

**Comment by ibm2431** [comment]
> Q: does unkillable prevent doom death and other instakills?

Looking like it doesn't. Unkillable just prevents your HP from falling down to 0 in `addHP`. Doom just calls `setHP(0)` outright. So you're still technically killable.

**Comment by TeoTwawki** [comment]
> 
> 
> > Q: does unkillable prevent doom death and other instakills?
> 
> Looking like it doesn't. Unkillable just prevents your HP from falling down to 0 in `addHP`. Doom just calls `setHP(0)` outright. So you're still technically killable.

K I still need doom screen and deathres if I test mobs with access to those then :( that sucks

**Comment by ibm2431** [comment]
Actually! I may have lied! I didn't read `setHP` very closely!

```cpp
    ((CBattleEntity*)m_PBaseEntity)->health.hp = 0;
    auto value = (int32)lua_tointeger(L, 1);
    ((CBattleEntity*)m_PBaseEntity)->addHP(value);
```

Since `addHP` was called and the entity's HP is currently zero, unkillable bumps it back up to 1 (in `addHP`).

Just tested with `!immortal` and `!hp 0`!

**Review by TeoTwawki — APPROVED** [review]
re-approved

**Pull request merged**